### ISSUE TITLE
Building/Compiling whole project

### ISSUE THREAD

**Issue by Springrbua** [angelozerr/typescript.java#156]
Currently, only saved files (and files on which the saved file depends) seem to be compiled with tsserver.  
Therefore you only see errors in those files. Errors in other files arn't visible, until you open and save that files to.  
This causes some problems, for example, when you are working with [angular2-eclipse](https://github.com/angelozerr/angular2-eclipse). When you create a new Angular2-Project, many ts-files are already created for you. They all contain Single-Quotes and Space-Indentation, which in my case causes tslint-errors.  
However, the tslint-errors are only visible, when I open and save the files.  
It would be cool, to have a compilation (similar to eclipse java compilation), which (conditionally) compiles the whole project on startup, and the saved file, when you edit inside Eclipse.  
On refresh, it might make sense to compile the whole project again.

**Comment by jcompagner** [comment]
you have to enabled in the tsconfig.json
"build on save" and "build on compile" (or at least one of them, don't know the exact difference) so that building goes a bit more automatically 
At least for me it will compile/transpile more.

But still it doesn't fully do it if i change the ts file outside of eclipse, and then do a refresh
The js file is not generated, until i open the file in the ts editor 
And then even if i first do a refresh then open the editor the js file is still not generated
only when the editor is already open and then the file is refreshed the js file is generated

So somehow the file change detection is really tied to the ts editor?

**Comment by Springrbua** [comment]
@jcompagner thanks for your tips, I'll try that.  
How is the performance, when selection "build on save"?  
In other plugins, it is recommanded to not use it, as it reduces performance a lot...

**Comment by jcompagner** [comment]
can you explain to me first what those 2 options in the json file really do?
Because i don't really notice any difference in behavior i think what ever those 2 settings are (good or bad)

What is the difference between the 2?

**Comment by angelozerr** [comment]
Indeed it's not clear. It depends the version of TypeScript Runtime:

 * with TypeScript < 2.0.5. The tsc command is used
  * compileOnSave executes tsc only for the ts saved file
  * buildOnSave execute tsc for the whole ts files of the project. It's helpful if they are ts files which depends on the saved ts files. But it can be very slowly.

 * with TypeScript >= 2.0.5. The tsserver is used to compile the saved ts file. They are a lot of benefit with that (fast performance)

In short answer buildOnSave is not used with TypeScript >= 2.0.5
